### PR TITLE
added instructiont::turn_into_skip()

### PR DESCRIPTION
**Pull request by kroening**
This is an implementation for the common idiom in which an existing
instruction is replaced by a skip while retaining the source_location.  It
replaces `instructiont::make_skip()`.

- [X] Each commit message has a non-empty body, explaining why the change was made.
- [X] Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- n/a The feature or user visible behaviour I have added or modified has been documented in the User Guide in doc/cprover-manual/
- [ ] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- n/a My commit message includes data points confirming performance improvements (if claimed).
- [X] My PR is restricted to a single feature or bugfix.
- n/a White-space or formatting changes outside the feature-related changed lines are in commits of their own.

### CODE REVIEW

**Review by allredj — COMMENTED** [review]
:heavy_check_mark:
Passed Diffblue compatibility checks (cbmc commit: 86d6fbe5653c079c66eb899a293d53e782e2e8cd).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/100492405

File: src/goto-programs/goto_program.h [hunk 1]
```diff
@@ -332,6 +332,13 @@ class goto_programt
       code.make_nil();
     }
 
+    /// Transforms an existing instruction into a skip,
+    /// retaining the source_location
+    void turn_into_skip()
+    {
+      clear(SKIP);
+    }
```
Comment (romainbrenguier) [inline src/goto-programs/goto_program.h:340]: `make_skip` exists which already does the same thing

Comment (kroening) [reply src/goto-programs/goto_program.h:340]: Indeed; however, the prefix `make_` is now commonly used for factories, and most programmers will be surprised to see these have an effect on `this`. My hope is that `turn_into_` makes this much clearer.

**Review by tautschnig — APPROVED** [review]

File: src/goto-programs/goto_program.h (hunk 1 above)
Comment (romainbrenguier) [reply src/goto-programs/goto_program.h:340]: In that case, `make_skip` should be removed (or at least deprecated), and the other `make_` functions should be renamed as well.

Comment (kroening) [reply src/goto-programs/goto_program.h:340]: Yes, coming!

**Pull request merged**